EIPS/eip-7697.md

```diff
@@ -0,0 +1,155 @@
+---
+eip: 7697
+title: AUTHCREATE opcode
+description: Validating and setting code for deterministic addresses.
+author: Xin Tian (@txgyy), Elim Poon (@yaonam)
+discussions-to: https://ethereum-magicians.org/t/eip-7697-authcreate-opcode/19780
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-04-23
+requires: 155
+---
+
+## Abstract
+
+The EIP introduces an EVM instruction `AUTHCREATE`. It allows a deterministic addresses to migrate to a smart contract.
+
+Inspiration comes from [EIP-3074](./eip-3074.md) and [EIP-5003](./eip-5003.md):
+
+- ([EIP-3074](./eip-3074.md)) + [EIP-5003](./eip-5003.md) = (`AUTH` + `AUTHCALL`) + `AUTHUSURP`
+- [EIP-7697](./eip-7697.md) = `AUTHCREATE` = `AUTH` + `AUTHUSURP`
```

Comment:
Linking to these proposals will prevent this proposal from advancing in status. Not a problem if that's what you want; just wanted to let you know.